docker/headnode.dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:20.04
+SHELL ["/bin/bash", "-c"]
+ENV HADOOP_HOME=/usr/local/hadoop/current
+ENV JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-amd64
+ENV HADOOP_LOG_DIR=/var/log/hadoop
+COPY headnode_start.sh /
+RUN apt update -y && \
```

Comment:
Make sense to separate "heavy" and "light" tasks.
For example:
```
RUN "some long-running steps"
RUN "some fast-running steps"
```
In that case, if you'll make changes after "some long-running steps", you'll not have to wait for these operations as the layer wasn't touched

Reply:
sorry for the long answer, done